### PR TITLE
modifies mlflow reporter to use MlflowClient to improve performance, …

### PR DESCRIPTION
**Pull request by ajslone**
…adds test for mlflow reporter

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -22,3 +22,21 @@ style = pep440
 versionfile_source = uv/_version.py
 versionfile_build = uv/_version.py
 tag_prefix =
+
```
Comment (ajslone) [inline setup.cfg:25]: squelch mypy for third-party packages

File: uv/mlflow/reporter.py [hunk 2]
```diff
@@ -26,16 +29,34 @@ class MLFlowReporter(b.AbstractReporter):
   """Reporter implementation that logs metrics to mlflow."""
 
   def __init__(self):
-    pass
+    self._client = mlf.tracking.MlflowClient()
```
Comment (ajslone) [inline uv/mlflow/reporter.py:30]: Using a client here is ~5x faster on my machine. From the mlflow code, it appears that this avoids recreating the sqlalchemy engine for each logging call.

Comment (sritchie) [reply uv/mlflow/reporter.py:30]: amazing!

File: uv/reporter/state.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def set_reporter(r: AbstractReporter) -> AbstractReporter:
 def active_reporter(r: AbstractReporter):
   old_reporter = _active_reporter
   globals()['_active_reporter'] = r
-  yield
+  yield r
```
Comment (sritchie) [inline uv/reporter/state.py:49]: 👍

**Review by sritchie — APPROVED** [review]

**Comment by sritchie** [comment]
Really nice work on this!

**Pull request merged**